perllib/FixMyStreet/Cobrand/Default.pm

```diff
@@ -553,7 +553,21 @@ has its reopening_disallowed flag set.
 =cut
 
 sub reopening_disallowed {
-    my ($self, $problem) = @_;
+    my ($self, $problem, $user) = @_;
+
+    # always prefer a cobranded body's own reopening restrictions
+    my $area_cobrand;
+    for my $body (values %{$problem->bodies}) {
+        #last if $area_cobrand = $body->get_cobrand_handler;
+        if ($body->get_cobrand_handler) {
+            $area_cobrand = $body->get_cobrand_handler;
+        }
+    }
+    my $value = $area_cobrand->call_hook('reopening_disallowed', $problem, $user);
```

Comment:
As you can see from the test failures, `$area_cobrand` could be undef at this point and then this will die.

I do see the rationale for doing something like this, but it's not something we've done before in Default.pm; generally if there is different behaviour required on .com as well as a cobrand, it has been put in FixMyStreet.pm. `get_cobrand_handler` requires `council_area_id` which is only used in UK cobrands anyway, so it's probably a bit clearer for any reusers. And then the code in FixMyStreet.pm could use `to_body_named` as some other functions do there:
```
my $c = $self->{c};
return 1 if $problem->to_body_named("Merton") && $c->user_exists && (!$c->user->from_body || $c->user->from_body->name ne "Merton Council");
return super();
```
Or something like that.